server/util/db/db.go

```diff
@@ -51,3 +57,10 @@ func GetConfiguredDatabase(c *config.Configurator) (*DBHandle, error) {
 	}
 	return nil, fmt.Errorf("No database configured -- please specify at least one in the config")
 }
+
+func (d *DBHandle) StartOfDayTimestamp(offset int) string {
+	if d.dialect == sqliteDialect {
```

Comment:
Can we do this all in go and remove the dialect-switching?

Reply:
Done